src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1529,6 +1529,12 @@ def _process_type_spec(self, parent, type_spec):
 
         elif isinstance(type_spec, Fortran2003.Declaration_Type_Spec):
             # This is a variable of derived type
+            if type_spec.children[0].lower() == "class":
+                # We don't yet support declarations that use 'class'
+                # TODO #1504 extend the PSyIR for this variable type.
+                raise NotImplementedError(
+                    f"Could not process {str(type_spec)} - declarations "
```

Comment:
The str() inside the f-string looks redundant. Is there any reason to have it?

Reply:
Not sure what happened there - removed it.